src/providers/EtherspotUiContextProvider.tsx

```diff
@@ -157,51 +156,41 @@ const EtherspotUiContextProvider = ({ children, chainId = 1 }: EtherspotUiContex
    * @returns Array[ISmartWalletAddress]
    */
   const getSmartWalletAddresses = async (): Promise<(ISmartWalletAddress)[]> => {
```

Comment:
I'm still thinking this versus `useEffect` + `setState` to always set `ISmartWalletAddress` in state and return without promise, wdyt

Reply:
when building react dapp user can rely on c smart wallet addresses changes stored in state which returned by `useEtherspotUi`

Reply:
or maybe taken example by new `useEtherspotBalances` add `useEtherspotAddresses`? thoughts

Reply:
I think let's replace this with `useEtherspotAddresses`